### PR TITLE
Rework help and usage for user-benchmark

### PR DESCRIPTION
**Pull request by portante**
And also pbench-collect-sysinfo.

Along way, fix output of the unit tests, and add unit tests for the
added behaviors to both scripts.

### CODE REVIEW

**Comment by portante** [comment]
@ndokos, @chaitanyaenr, please review when you have a chance.  I'd like to have this merged sooner than later, and definitely ahead of the ara work.

File: agent/bench-scripts/pbench-user-benchmark [hunk 1]
```diff
@@ -69,15 +73,13 @@ while true; do
...
-				:
-			else
-				exit 1
-			fi
 			shift;
                 fi
                 ;;
+		-h|--help)
+		usage
+		exit 0;
```
Comment (chaitanyaenr) [inline agent/bench-scripts/pbench-user-benchmark]: ; is not necessary though it doesn't really matter. Everything else looks good to me.

File: agent/bench-scripts/gold/pbench-user-benchmark/test-07.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
++++ Running test-07 pbench-user-benchmark
+Usage: pbench-user-benchmark [options] -- <script to run>
+
+	The following options are available:
+
+		-C str --config=str            name of the test config
+		       --tool-group=str
+		       --sysinfo=str,          str = comma seperated values of sysinfo to be collected
```
Comment (ndokos) [inline agent/bench-scripts/gold/pbench-user-benchmark/test-07.txt]: separated

File: agent/bench-scripts/gold/pbench-user-benchmark/test-08.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	pbench-user-benchmark: you specified an invalid --sysinfo option ("bad")
+
+Usage: pbench-user-benchmark [options] -- <script to run>
+
+	The following options are available:
+
+		-C str --config=str            name of the test config
+		       --tool-group=str
+		       --sysinfo=str,          str = comma seperated values of sysinfo to be collected
```
Comment (ndokos) [inline agent/bench-scripts/gold/pbench-user-benchmark/test-08.txt]: Ditto

File: agent/bench-scripts/pbench-user-benchmark [hunk 4]
```diff
@@ -34,17 +34,21 @@ config=""
...
-if [ $? -ne 0 ]; then
-	printf -- "$*\n"
-	printf "\n"
-	printf "\t${benchmark}: you specified an invalid option\n\n"
+function usage {
+	printf "Usage: $script_name [options] -- <script to run>\n\n"
 	printf "\tThe following options are available:\n\n"
 	printf -- "\t\t-C str --config=str            name of the test config\n"
 	printf -- "\t\t       --tool-group=str\n"
 	printf -- "\t\t       --sysinfo=str,          str = comma seperated values of sysinfo to be collected\n"
```
Comment (ndokos) [inline agent/bench-scripts/pbench-user-benchmark]: Ditto

File: agent/util-scripts/gold/pbench-collect-sysinfo/test-23.txt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Usage: pbench-collect-sysinfo [options]
+
+Options specified can be one of:
+	-d str --dir=str,     str = a directory where the pbench-collect-sysinfo
+	                            will store and process data
+
+	-g str --group=str,   str = a tool group used in a benchmark
+	                            (the default group is 'default')
+
+	       --sysinfo=str, str = comma seperated values of sysinfo to be collected
```
Comment (ndokos) [inline agent/util-scripts/gold/pbench-collect-sysinfo/test-23.txt]: Ditto

File: agent/util-scripts/pbench-collect-sysinfo [hunk 6]
```diff
@@ -28,20 +28,27 @@ function display_sysinfo_opts {
...
-	printf "$script_name: you specified an invalid option\n\n"
-	printf "The following are required:\n\n"
+	printf "Options specified can be one of:\n"
 	printf -- "\t-d str --dir=str,     str = a directory where the $script_name\n"
 	printf -- "\t                            will store and process data\n"
 	printf "\n"
 	printf -- "\t-g str --group=str,   str = a tool group used in a benchmark\n"
 	printf -- "\t                            (the default group is 'default')\n"
+	printf "\n"
 	printf -- "\t       --sysinfo=str, str = comma seperated values of sysinfo to be collected\n"
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Ditto

**Comment by portante** [comment]
What do you mean about misleading output, @ndokos?

**Comment by ndokos** [comment]
In the output below, `pbench-collect-sysinfo` complains about an invalid option:

    # ./pbench-user-benchmark --help
    Usage: pbench-user-benchmark [options] -- <script to run>

            The following options are available:

                    -C str --config=str            name of the test config
                           --tool-group=str
                           --sysinfo=str,          str = comma seperated values of sysinfo to be collected
                                                         available: Collecting system information

    pbench-collect-sysinfo: you specified an invalid option

    The following are required:

            -d str --dir=str, str = a directory where the pbench-collect-sysinfo
                                    will store and process data

            -g str --group=str, str = a tool group used in a benchmark
                                      (the default group is 'default')

**Comment by portante** [comment]
That is because you need to use the pbench-collect-sysinfo that is in tree, I believe.

**Comment by ndokos** [comment]
I 'll check again but I had set my PATH=.:$PATH so it should pick the tree one.

**Comment by portante** [comment]
@ndokos, @chaitanyaenr, replace double check and merge when appropriate.

**Pull request merged**